worker/src/worker.rs

```diff
@@ -294,8 +294,10 @@ struct WorkerReceiverHandler<PublicKey: VerifyingKey> {
 }
 
 impl<PublicKey: VerifyingKey> WorkerReceiverHandler<PublicKey> {
-    fn spawn(self, address: SocketAddr) {
+    fn spawn(self, address: SocketAddr, max_concurrent_requests: usize) {
+        let concurrent_limit = tower::limit::ConcurrencyLimitLayer::new(max_concurrent_requests);
         let service = tonic::transport::Server::builder()
+            .layer(concurrent_limit)
```

Comment:
Instead of building a `ConcurrencyLimitLayer` we can actually use the [concurrency_limit_per_connection](https://docs.rs/tonic/latest/tonic/transport/server/struct.Server.html#method.concurrency_limit_per_connection) function on the Server builder directly, assuming i'm understanding the semantics you're after

Reply:
That's interesting. I thought that the `ConcurrencyLimitLayer` will be applied across all connections - not per connection (but per `service` though). @laura-makdah I think it would help clarifying the intention so we make sure we are aligned here.